doc/neps/nep-0035-array-creation-dispatch-with-array-function.rst

```diff
@@ -8,16 +8,154 @@ NEP 35 — Array Creation Dispatching With __array_function__
+defined by the downstream library implementing that type, which also implements
+the ``__array_function__`` protocol. With this we address one of that
+protocol's shortcomings, as described by NEP 18 [1]_.
+
+Motivation and Scope
+--------------------
+
+Many are the libraries implementing the NumPy API, such as Dask for graph
+computing, CuPy for GPGPU computing, xarray for N-D labeled arrays, etc. All
+the libraries mentioned have yet another thing in common: they have also adopted
+the ``__array_function__`` protocol. The protocol defines a mechanism allowing a
+user to directly use the NumPy API as a dispatcher based on the input array
+type. In essence, dispatching means users are able to pass a downstream array,
+such as a Dask array, directly to one of NumPy's compute functions, and NumPy
+will be able to automatically recognize that and send the work back to Dask's
+implementation of that function, which will define the return value. For
+example:
+
+.. code:: python
+
+    x = dask.array.arange(5)    # Creates dask.array
+    np.sum(a)                   # Returns dask.array
+
+Note above how we called Dask's implementation of ``sum`` via the NumPy
+namespace by calling ``np.sum``, and the same would apply if we had a CuPy
+array or any other array from a library that adopts ``__array_function__``.
+This allows writing code that is agnostic to the implementation library, thus
+users can write their code once and still be able to use different array
+implementations according to their needs.
+
+Unfortunately, ``__array_function__`` has limitations, one of them being array
+creation functions. In the example above, NumPy was able to call Dask's
+implementation because the input array was a Dask array. The same is not true
+for array creation functions, in the example the input of ``arange`` is simply
+the integer ``5``, not providing any information of the array type that should
+be the result, that's where a reference array passed by the ``like=`` argument
+proposed here can be of help, as it provides NumPy with the information
+required to create the expected type of array.
+
+The new ``like=`` keyword proposed is solely intended to identify the downstream
+library where to dispatch and the object is used only as reference, meaning that
+no modifications, copies or processing will be performed on that object.
+
+We expect that this functionality will be mostly useful to library developers,
+allowing them to create new arrays for internal usage based on arrays passed
+by the user, preventing unnecessary creation of NumPy arrays that will
+ultimately lead to an additional conversion into a downstream array type.
+
+Support for Python 2.7 has been dropped since NumPy 1.17, therefore we make use
+of the keyword-only argument standard described in PEP-3102 [2]_ to implement
+``like=``, thus preventing it from being passed by position.
+
+.. _neps.like-kwarg.usage-and-impact:
+
+Usage and Impact
```

Comment:
This section is a nice read much clearer than what I tried to understand. Would it be possible to give a few sentences for the regular NumPy users who won't use this feature?

Reply:
I mean what they would see and possibly avoid if this is implemented.

Reply:
I added a sentence at the end, but to be honest, I'm not sure if that's what you meant. Let me know if that addresses your question or if I completely misunderstood your suggestion. :)